fe/fe-core/src/main/java/com/starrocks/catalog/Database.java

```diff
@@ -298,6 +298,18 @@ public boolean registerTableUnlocked(Table table) {
         }
     }
 
+    public boolean registerTemporaryTableUnlocked(Table table) {
```

Comment:
can we merge these two methods , 
![image](https://github.com/StarRocks/starrocks/assets/107105845/081c0625-9989-4b23-8c3b-2125c5589e9e)
so that we don't need to have logic like this :
![image](https://github.com/StarRocks/starrocks/assets/107105845/8609614c-60eb-49a1-8e31-4f3c8a620a00)
and this:
![image](https://github.com/StarRocks/starrocks/assets/107105845/4d9b932c-58fe-42a8-a941-d718c2811706)
they are almost the same

Reply:
fixed, I've merge these logic to registerTableUnlocked and remove registerTemporaryTableUnlocks